service/controller/v2/resource/node/create.go

```diff
@@ -114,30 +123,58 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	if len(customPods) > 0 {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting all pods running custom workloads")
+		t1 := time.Now()
 
+		var wg sync.WaitGroup
 		for _, p := range customPods {
-			err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
-			if err != nil {
-				return microerror.Mask(err)
-			}
+			// delete each pod in separate goroutine to avoid blocking
+			wg.Add(1)
+			go func(k8sClient kubernetes.Interface, pod v1.Pod) {
+				defer wg.Done()
+				o := func() error {
+					return EvictPod(k8sClient, pod)
+				}
+				b := backoff.NewExponential(backoff.ShortMaxWait, backoff.ShortMaxInterval)
+				n := backoff.NewNotifier(r.logger, ctx)
+				err = backoff.RetryNotify(o, b, n)
```

Comment:
I am not sure we should retry here, because we have the retry resource. @kopiczko what is your idea here? How should that be at best?

Reply:
I think short ResyncPeriod and aggressive cancelling on error. I like your suggestion about `errgroup`. I think nodeconfigs are short living objects so we can afford listing them often.

Blocking for 2 min (ShortMaxWait) doesn't make any sense to me. We can just set ResyncPeriod to 2min. Only backoff that would make sense here IMO is just `NewMaxRetries`. Something like `backoff.MaxRetries(3, 1*time.Second)` especially when this is done in concurrent fashion. We can spend like couple of seconds in the loop in the worst case scenario I think.